src/Microsoft.DocAsCode.Dfm/DfmCodeExtractor.cs

```diff
@@ -11,10 +11,41 @@ namespace Microsoft.DocAsCode.Dfm
     using Microsoft.DocAsCode.Common;
     using Microsoft.DocAsCode.Plugins;
 
-    internal class DfmCodeExtractor
+    public class DfmCodeExtractor
     {
         private static readonly string RemoveIndentSpacesRegexString = @"^[ \t]{{1,{0}}}";
 
+        private readonly IDfmFencesBlockPathQueryOptionCreator[] _pathQueryOptionCreaters;
+
+        public DfmCodeExtractor(IDfmFencesBlockPathQueryOptionCreator[] pathQueryOptionCreaters = null)
+        {
+            _pathQueryOptionCreaters = pathQueryOptionCreaters ?? new IDfmFencesBlockPathQueryOptionCreator[]
+            {
+                new FullFileBlockPathQueryOptionCreater(),
+                new TagNameBlockPathQueryOptionCreater(),
+                new MultipleLineRangeBlockPathQueryOptionCreater(),
+            };
+        }
+
+        public IDfmFencesBlockPathQueryOption ParsePathQueryString(string queryOrFragment, bool noCache = false)
+        {
+            if (string.IsNullOrEmpty(queryOrFragment))
+            {
+                return null;
+            }
+
+            var parameters = DfmFencesBlockPathQueryOptionParameters.Create(queryOrFragment);
+            foreach (var creater in _pathQueryOptionCreaters)
```

Comment:
One creator to contain all the options and move this logic to the creator?

Reply:
sounds nice!

---
In reply to: [147903380](https://github.com/dotnet/docfx/pull/2163#discussion_r147903380) [](ancestors = 147903380)